server/apps/users/helpers.py

```diff
@@ -38,3 +38,18 @@ def user_submitted_profile(user):
         submitted = False
     return submitted
 
+def get_user_profile(user):
+    """
+    Attempts to get the profile for a user.
+
+    Args:
+        user: request.user
+
+    Returns:
+        The UserProfile associated with the user if it exists, None otherwise
+    """
+    try:
+        uo = UserProfile.objects.get(user=user)
+    except:
```

Comment:
Minor comment, could we change this to be a more specific exception than the blanket `except`? Not too relevant as the only other thing than `DoesNotExist` would be if it returned >1 item, but still nicer to keep with the Python recommendations! 🙂

Reply:
Yes, good point. I've refined it to just catch the `DoesNotExist` exception. We could also include the [`MultipleObjectsReturned`](https://docs.djangoproject.com/en/4.2/ref/models/class/#django.db.models.Model.MultipleObjectsReturned) case but I can't envisage a situation where this should happen for any good reason, so think it's best to leave it out so we know to look into it if it hapens. Does that seem okay to you?

Reply:
Absolutely, that's very sensible! 👍